docs/reactivity/integrate-backend-api.md

```diff
@@ -0,0 +1,77 @@
+# Building a reactive frontend with Oryx
+
+## Qualify the request
+
+Components, services and adapters are customizable. To ensure that a change in the requested data by a component can be used in lower layers, a best practice is to use a so-called qualifier in the application flow. The qualifier is a single object that will be passed fro application layer to application layer, and can be extended. The example below shows the use of the `ProductQualifier` in the ProductService.
+
+```ts
+export interface ProductQualifier {
+  sku?: string;
+  include?: string[];
+}
+
+export class DefaultProductService implements ProductService {
+  get(qualifier: ProductQualifier): Observable<NullableGeneric<Product>> {
+    return this.getProductData(qualifier).value;
+  }
+}
+```
+
+Whenever the qualifier must be extended, say we like to query the product for a certain brand or supplier, the qualifier can be extended:
+
+```ts
+declare global {
+  interface ProductQualifier {
+    brand?: string;
+  }
+}
+```
+
+This change will be transparent to all application layers and can be picked up by the Adapter layer to use the `brand` field to fetch products by brand.
+
+## Designing the data model
```

Comment:
This paragraph talks a lot about a client model. Should we rename the title? Or how client and data models are related?

Reply:
it is about integrating backend APIs. The client model is part of it; when you adapt a 3rd party API, you must ensure that you convert the (3rd party) API response to the client model, otherwise other application layers and the components will stop working. 

So, it's an important topic, but not the main topic.

Reply:
Either way, I've changed the doc a bit.